LibGfx: Enable explicit symbol export

R-Goc opened the pull request:
This commit enables explicit symbol export for LibGfx. This decreases the number of exported symbols from around ~1500 to ~400.

ADKaster commented on Libraries/LibGfx/Path.h:
```diff
@@ -9,14 +9,15 @@
 namespace Gfx {
 
-class PathImpl {
+class GFX_API PathImpl {
```
same here, does Path*Impl* need exported?

R-Goc replied:
I'll retry the Impl ones to make sure.

R-Goc commented on Libraries/LibGfx/Palette.h:
```diff
@@ -12,11 +12,12 @@
 namespace Gfx {
 
-class PaletteImpl : public RefCounted<PaletteImpl> {
+class GFX_API PaletteImpl : public RefCounted<PaletteImpl> {
```
PaletteImpl is directly used in WebContent

R-Goc committed:
AK: Properly export LSAN suppressions

When enabling export macros for LibGfx the symbols to suppress lsan
warnings were not getting exported. As such this change makes sure that
the symbols get exported when the library has hidden symbol visibility
as default.

R-Goc committed:
LibGfx: Enable explicit symbol export

This commit enables explicit symbol export for LibGfx. This decreases
the number of exported symbols from around ~1500 to ~400.

R-Goc committed:
Meta: Remove ENABLE_GUI_TARGETS

This commit removes the ENABLE_GUI_TARGETS option as it is not
necessary.
